src/deadline/job_attachments/_diff.py

```diff
@@ -117,47 +118,184 @@ def compare_manifest(
 
 
 def _fast_file_list_to_manifest_diff(
-    root: str, current_files: List[str], diff_manifest: BaseAssetManifest, logger: ClickLogger
-) -> List[str]:
+    root: str,
+    current_files: List[str],
+    diff_manifest: BaseAssetManifest,
+    logger: ClickLogger,
+    return_root_relative_path: bool = True,
+) -> List[Tuple[str, FileStatus]]:
     """
     Perform a fast difference of the current list of files to a previous manifest to diff against using time stamps and file sizes.
     :param root: Root folder of files to diff against.
     :param current_files: List of files to compare with.
     :param diff_manifest: Manifest containing files to diff against.
+    :param return_root_relative_path: File Path to return, either relative to root or full.
+    :param logger: logger.
     :return List[str]: List of files that are new, or modified.
     """
-    changed_paths: List[str] = []
+
+    def select_path(full_path: str, relative_path: str, return_root_relative_path: bool):
```

Comment:
Q - what's the use case for full or relative path?

Reply:
in the snapshot case, it requires the full path. Snapshot (and related code) uses the full path. Diff on the otherhand uses relative path. I didn't want to duplicate the code, so I used a function toggle to allow returning in either format easily.